### PR TITLE
cloud computing sol

### PR DESCRIPTION
**Pull request by SansPapyrus683**
someone please check the editorial

### CODE REVIEW

File: solutions/ceoi18-clo.mdx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	std::sort(
+		poss_transactions.begin(), poss_transactions.end(),
+		[] (const Transaction& a, const Transaction& b) -> bool {
+			return a.rate != b.rate ? a.rate > b.rate : a.price < b.price;
+		}
+	);
+
+	/*
+	 * max_profits[t][c] = the maximum profit we can gain from the first
+	 * t transactions given that we have $c$ cores left
```
Comment (nwatx) [inline solutions/ceoi18-clo.mdx:165]: is there a reason why t isn't texified but c is

Comment (SansPapyrus683) [reply solutions/ceoi18-clo.mdx:165]: oh
oh wait
what why did i even put that there

File: solutions/ceoi18-clo.mdx [hunk 2]
```diff
@@ -0,0 +1,185 @@
+---
+id: ceoi18-clo
+source: CEOI 18
+title: Cloud Computing
+author: Kevin Sheng
+---
+
+It might be tempting at first to do a 2-dimensional DP with the number of computer purchases and the number of customer orders, but that would result in a nasty and inefficient transition.
+
+Instead, let's try grouping the two into a single array of "transactions". Each transation has three attributes- the change in the number of cores, the associated clock rate of the cores, and the change in the total profits.
```
Comment (caoash) [inline solutions/ceoi18-clo.mdx:10]: lol can you replace - again

File: solutions/ceoi18-clo.mdx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+			<td>1500</td>
+			<td>300</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Notice that in this order, as long as a customer order comes after a computer purchase, the computer will be able to satisfy the order as long as there are enough cores.
+
+This drastically simplifies the problem. Now we can simply define $\texttt{max_profits}[t][c]$ as the maximum profit we can gain from the first $t$ transactions given that we have $c$ cores left.
```
Comment (caoash) [inline solutions/ceoi18-clo.mdx:111]: can you be more specific what the simplified problem is

**Review by caoash — CHANGES_REQUESTED** [review]
ping me after changes, ill merge

File: solutions/ceoi18-clo.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/ceoi18-clo.mdx:111]: i mean it's just knapsack at this point it's kinda obvious
i also say it later too

**Review by ryanchou-dev — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
@caoash

**Comment by ryanchou-dev** [comment]
> @caoash

maybe ping him on discord? he's online rn

**Pull request merged**